paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -372,16 +374,23 @@ def mesos_cpu_metrics_provider(
             last_time = 0.0
             last_cpu_data = []
 
-    monkey.patch_socket()
-    jobs = [gevent.spawn(task.stats_callable) for task in mesos_tasks]
-    gevent.joinall(jobs, timeout=60)
-    mesos_tasks = dict(zip([task['id'] for task in mesos_tasks], [job.value for job in jobs]))
+    futures = [asyncio.ensure_future(task.stats()) for task in mesos_tasks]
+    if futures:
+        a_sync.block(asyncio.wait, futures, timeout=60)
+
+    def results_or_None(fut):
+        try:
+            return fut.result()
+        except Exception:
```

Comment:
not a fan of using exceptions for control flow, how about this:

    if fut.exception():
        return None
    else
        return fut.result()

this may still raise `CancelledError` or `InvalidStateError` - but those could signify actual bug in the code (useful for debugging) and/or maybe we want to do something about it here or higher up the stack?